src/Parser/EnhancementShaman/Modules/ShamanCore/Pandemic.js

```diff
@@ -0,0 +1,58 @@
+import SPELLS from 'common/SPELLS';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Module from 'Parser/Core/Module';
+
+const SPELLID = SPELLS.FLAMETONGUE_BUFF.id;
+const PANDEMIC_THRESHOLD = 11500;
+const debug = false;
+
+class Pandemic extends Module {
+  ft_timestamp = [];
+  early_refresh = [];
+  refresh_avearge;
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  on_byPlayer_applybuff(event) {
+    if(event.ability.guid === SPELLID) {
+      this.ft_timestamp.push(event.timestamp);
+
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    if(event.ability.guid === SPELLID) {
+      this.ft_timestamp.push(event.timestamp);
+    }
+  }
+
+  on_finished() {
+    let diff;
+
+    for (let i = 1; i < this.ft_timestamp.length; i++) {
+      diff = this.ft_timestamp[i] - this.ft_timestamp[i - 1];
+      if(diff < PANDEMIC_THRESHOLD) {
+        this.early_refresh.push(diff);
```

Comment:
It's generally preferred to do the actual data fetching in the related events, in this case `on_byPlayer_refreshbuff` instead of storing a buffer and processing it in on_finished. Instead of tracking the array `ft_timestamp` you would instead track something like the `lastFlameTongueAppliedTimestamp` which would be a simpler value.

Also this will probably break if FT is cast pre-combat.